yt_dlp/extractor/dropbox.py

```diff
@@ -56,8 +56,8 @@ def _real_extract(self, url):
             else:
                 raise ExtractorError('Password protected video, use --video-password <password>', expected=True)
 
-        json_string = self._html_search_regex(r'InitReact\.mountComponent.+ "props":(.+), "elem_id"', webpage, 'Info JSON')
-        info_json = self._parse_json(json_string, video_id)
+        json_string = self._html_search_regex(r'InitReact\.mountComponent\(.*?,\s*(\{.+\})\s*?\)', webpage, 'Info JSON')
```

Comment:
The page may contain more than one `InitReact.mountComponent(...)`. The one with `"module_name": "metaserver/static/js/modules/clean/react/file_viewer/container_with_skeleton"` is the right one. If the first one found is not that one, the extractor crashes with `transcode_url` set to `None`.
```suggestion
        json_string = next(
            x for x in (m.group(1) for m in re.finditer(
                r'(?s)InitReact\s*\.\s*mountComponent\s*\(.*?,\s*(\s*\{.+?\})\s*?\)', webpage))
            if '/react/file_viewer/' in x)
```